---
title: "rphylotastic package"
author: "Brian O'Meara and Luna L. SÃ¡nchez-Reyes"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{rphylotastic usage}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

`rphylotastic` connects to Phylotastic web services, which themselves wrap and process multiple other services as well as have unique services of their own.

# Getting trees

The main use for `rphylotastic` is getting phylogenetic trees for a set of taxa of interest. These can come from Open Tree of Life (topology only, many polytomies), Phylomatic (has branch lengths, but only plants), or NCBI taxonomy (even more polytomies than Open Tree's tree). See also the `treebase` and `rotl` packages for getting individual trees, or `datelife` for getting chronograms (a single synthetic one or individual ones).

```{r, fig.align = 'center', fig.width = 7}
library(rphylotastic)
taxon_names <- c("Crabronidae", "Ophiocordyceps", "Megalyridae",
          "Formica polyctena", "Tetramorium caespitum",
          "Pseudomyrmex", "Carebara diversa", "Formicinae")

# phy1 <- taxa_get_otol_tree(taxa = taxon_names)
#
# phy2 <- taxa_get_phylomatic_tree(taxa = taxon_names)
#
# phy3 <- taxa_get_taxonomic_tree(taxa = taxon_names)

source_tree <- c("Open Tree of Life", "Phylomatic", "taxonomic tree")
library(ape)
# par(xpd = FALSE, mfrow = c(1, 3), mai = c(0.1, 0.1, 1, 0.1), oma = c(0, 0, 3, 0))
# for (i in c(phy1, phy2, phy3)){
# for (i in 1:3){
#   plot(i, cex = 0.5, label.offset = 0.1)
#   mtext(paste("from", source_tree[i]), cex = 2)
# }
# mtext("Some ant species trees", outer = T, side = 3, cex = 3)

```
With `rphylotastic`, you can also check if trees are equal
```{r}
# phylo_compare(phy1, phy2)
# phylo_compare(phy2, phy3)
```

# Getting scientific names
## From scratch
One very cool feature of `rphylotastic` is the ability to scrape scientific names from documents or web pages. For example, you can get all latin names from a wikipedia article on primates:

```{r}
sci_names <- url_get_scientific_names(URL = "https://en.wikipedia.org/wiki/Primate")
length(sci_names)
head(sci_names)
```

You can also extract all scientific names from a text you are studying:
```{r}
sci_names <- text_get_scientific_names(text = "The domestic cat (Felis silvestris catus or Felis catus) is a small, typically furry, carnivorous mammal.")
head(sci_names)
```


## From a taxon of interest
If you are interested in a particular group, and you want to get a tree containing all species within the group, you can call `taxon_get_species` function to get a vector of all species within the group registered in Open Tree of Life taxonomy, and then use that vector to perform a tree search.

```{r}
ant_species <- taxon_get_species(taxon = "Formicidae")
more_species <- taxon_get_species(taxon = c("Formicidae", "Felidae"))
length(more_species)
length(ant_species)
head(ant_species)
# phy1 <- taxa_get_otol_tree(taxa = ant_species)
```
You can only use this function one taxon at a time
```{r}
more_species <- taxon_get_species(taxon = c("Formicidae", "Felidae"))
length(more_species)
```
But you can use a handy apply function
```{r}
more_species <- rapply(list("Formicidae", "Felidae"), taxon_get_species)
length(more_species)
```
Or see `datelife` package for a wrapper of this function that accepts multiple taxon names at a time.

### Subsetting species names
Species from a taxon can also be subset by country or by genome:
```{r}
spp_from_country <- taxon_get_species_from_country("Formicidae", "Mexico")
length(spp_from_country)
head(spp_from_country)

spp_with_genome <- taxon_get_species_with_genome("Formicidae")
length(spp_with_genome)
head(spp_with_genome)
```
# Checking your scientific names
You may already have a list of names, but you want to check if they all are good scientific names. `rphylotastic` calls Global Name Resolver and the Taxonomic Name Resolution Service to verify names. This will keep correct names only.

```{r}
species <- c("Felis silvestris", "Home sepiens")
taxa_resolve_names_with_otol(taxa = species)
```

# Getting data from Encyclopedia of Life
You can also get some images:

```{r}
some_images <- species_get_image_data(species = c("Homo sapiens", "Felis catus"))
```
Or other miscelanous info:
```{r}
some_info <- species_get_info(species = c("Homo sapiens", "Felis catus"))
```


# Citing

The phylotastic manuscript is still being written.
