<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Brian O’Meara and Luna L. Sánchez-Reyes" />

<meta name="date" content="2018-06-18" />

<title>rphylotastic package</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>



<link href="data:text/css;charset=utf-8,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23header%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%20code%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" rel="stylesheet" type="text/css" />

</head>

<body>




<h1 class="title toc-ignore">rphylotastic package</h1>
<h4 class="author"><em>Brian O’Meara and Luna L. Sánchez-Reyes</em></h4>
<h4 class="date"><em>2018-06-18</em></h4>



<p><code>rphylotastic</code> connects to Phylotastic web services, which themselves wrap and process multiple other services as well as have unique services of their own.</p>
<div id="getting-trees" class="section level1">
<h1>Getting trees</h1>
<p>The main use for <code>rphylotastic</code> is getting phylogenetic trees for a set of taxa of interest. These can come from Open Tree of Life (topology only, many polytomies), Phylomatic (has branch lengths, but only plants), or NCBI taxonomy (even more polytomies than Open Tree’s tree). See also the <code>treebase</code> and <code>rotl</code> packages for getting individual trees, or <code>datelife</code> for getting chronograms (a single synthetic one or individual ones).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(rphylotastic)
<span class="kw">library</span>(datelife)
taxon_names &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Crabronidae&quot;</span>, <span class="st">&quot;Ophiocordyceps&quot;</span>, <span class="st">&quot;Megalyridae&quot;</span>,
          <span class="st">&quot;Formica polyctena&quot;</span>, <span class="st">&quot;Tetramorium caespitum&quot;</span>,
          <span class="st">&quot;Pseudomyrmex&quot;</span>, <span class="st">&quot;Carebara diversa&quot;</span>, <span class="st">&quot;Formicinae&quot;</span>)

<span class="co"># phy1 &lt;- taxa_get_otol_tree(taxa = taxon_names)</span>
<span class="co"># </span>
<span class="co"># phy2 &lt;- taxa_get_phylomatic_tree(taxa = taxon_names)</span>
<span class="co"># </span>
<span class="co"># phy3 &lt;- taxa_get_taxonomic_tree(taxa = taxon_names)</span>

source_tree &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Open Tree of Life&quot;</span>, <span class="st">&quot;Phylomatic&quot;</span>, <span class="st">&quot;taxonomic tree&quot;</span>)
<span class="kw">library</span>(ape)
<span class="co">#&gt; Warning: package 'ape' was built under R version 3.4.4</span>
<span class="co"># par(xpd = FALSE, mfrow = c(1, 3), mai = c(0.1, 0.1, 1, 0.1), oma = c(0, 0, 3, 0))</span>
<span class="co"># for (i in c(phy1, phy2, phy3)){</span>
<span class="co"># for (i in 1:3){</span>
<span class="co">#   plot(i, cex = 0.5, label.offset = 0.1)</span>
<span class="co">#   mtext(paste(&quot;from&quot;, source_tree[i]), cex = 2)</span>
<span class="co"># }</span>
<span class="co"># mtext(&quot;Some ant species trees&quot;, outer = T, side = 3, cex = 3)</span></code></pre></div>
<p>With <code>rphylotastic</code>, you can also check if trees are equal</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># phylo_compare(phy1, phy2)</span>
<span class="co"># phylo_compare(phy2, phy3)</span></code></pre></div>
</div>
<div id="getting-scientific-names" class="section level1">
<h1>Getting scientific names</h1>
<div id="from-scratch" class="section level2">
<h2>From scratch</h2>
<p>One very cool feature of <code>rphylotastic</code> is the ability to scrape scientific names from documents or web pages. For example, you can get all latin names from a wikipedia article on primates:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sci_names &lt;-<span class="st"> </span><span class="kw">url_get_scientific_names</span>(<span class="dt">URL =</span> <span class="st">&quot;https://en.wikipedia.org/wiki/Primate&quot;</span>)
<span class="kw">length</span>(sci_names)
<span class="co">#&gt; [1] 169</span>
<span class="kw">head</span>(sci_names)
<span class="co">#&gt; [1] &quot;Daubentoniidae&quot; &quot;Tarsiidae&quot;      &quot;Lemuridae&quot;      &quot;Lorisidae&quot;     </span>
<span class="co">#&gt; [5] &quot;Cebidae&quot;        &quot;Callitrichidae&quot;</span></code></pre></div>
<p>You can also extract all scientific names from a text you are studying:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sci_names &lt;-<span class="st"> </span><span class="kw">text_get_scientific_names</span>(<span class="dt">text =</span> <span class="st">&quot;The domestic cat (Felis silvestris catus or Felis catus) is a small, typically furry, carnivorous mammal.&quot;</span>)
<span class="kw">head</span>(sci_names)
<span class="co">#&gt; [1] &quot;Felis silvestris catus&quot; &quot;Felis catus&quot;</span></code></pre></div>
</div>
<div id="from-a-taxon-of-interest" class="section level2">
<h2>From a taxon of interest</h2>
<p>If you are interested in a particular group, and you want to get a tree containing all species within the group, you can call <code>taxon_get_species</code> function to get a vector of all species within the group registered in Open Tree of Life taxonomy, and then use that vector to perform a tree search.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ant_species &lt;-<span class="st"> </span><span class="kw">taxon_get_species</span>(<span class="dt">taxon =</span> <span class="st">&quot;Formicidae&quot;</span>)
more_species &lt;-<span class="st"> </span><span class="kw">taxon_get_species</span>(<span class="dt">taxon =</span> <span class="kw">c</span>(<span class="st">&quot;Formicidae&quot;</span>, <span class="st">&quot;Felidae&quot;</span>))
<span class="kw">length</span>(more_species)
<span class="co">#&gt; [1] 10883</span>
<span class="kw">length</span>(ant_species)
<span class="co">#&gt; [1] 10883</span>
<span class="kw">head</span>(ant_species)
<span class="co">#&gt; [1] &quot;Hypopomyrmex bombiccii&quot;  &quot;Stigmomyrmex venustus&quot;  </span>
<span class="co">#&gt; [3] &quot;Imhoffia pallida&quot;        &quot;Imhoffia nigra&quot;         </span>
<span class="co">#&gt; [5] &quot;Kotshkorkia laticeps&quot;    &quot;Poneropterus sphecoides&quot;</span>
<span class="co"># phy1 &lt;- taxa_get_otol_tree(taxa = ant_species)</span></code></pre></div>
<p>You can only use this function one taxon at a time</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">more_species &lt;-<span class="st"> </span><span class="kw">taxon_get_species</span>(<span class="dt">taxon =</span> <span class="kw">c</span>(<span class="st">&quot;Formicidae&quot;</span>, <span class="st">&quot;Felidae&quot;</span>))
<span class="kw">length</span>(more_species)
<span class="co">#&gt; [1] 10883</span></code></pre></div>
<p>But you can use a handy apply function</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">more_species &lt;-<span class="st"> </span><span class="kw">rapply</span>(<span class="kw">list</span>(<span class="st">&quot;Formicidae&quot;</span>, <span class="st">&quot;Felidae&quot;</span>), taxon_get_species)
<span class="kw">length</span>(more_species)
<span class="co">#&gt; [1] 10982</span></code></pre></div>
<p>Or see <code>datelife</code> package for a wrapper of this function that accepts multiple taxon names at a time.</p>
<div id="subsetting-species-names" class="section level3">
<h3>Subsetting species names</h3>
<p>Species from a taxon can also be subset by country or by genome:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">spp_from_country &lt;-<span class="st"> </span><span class="kw">taxon_get_species_and_country</span>(<span class="st">&quot;Formicidae&quot;</span>, <span class="st">&quot;Mexico&quot;</span>)
<span class="kw">length</span>(spp_from_country)
<span class="co">#&gt; [1] 479</span>
<span class="kw">head</span>(spp_from_country)
<span class="co">#&gt; [1] &quot;Acanthoponera goeldii&quot; &quot;Acanthoponera minor&quot;   &quot;Tatuidris tatusia&quot;    </span>
<span class="co">#&gt; [4] &quot;Amblyopone orizabana&quot;  &quot;Prionopelta modesta&quot;   &quot;Prionopelta amabilis&quot;</span>

spp_with_genome &lt;-<span class="st"> </span><span class="kw">taxon_get_species_with_genome</span>(<span class="st">&quot;Formicidae&quot;</span>)
<span class="kw">length</span>(spp_with_genome)
<span class="co">#&gt; [1] 26</span>
<span class="kw">head</span>(spp_with_genome)
<span class="co">#&gt; [1] &quot;Ooceraea biroi&quot;         &quot;Harpegnathos saltator&quot; </span>
<span class="co">#&gt; [3] &quot;Dinoponera quadriceps&quot;  &quot;Aphaenogaster picea&quot;   </span>
<span class="co">#&gt; [5] &quot;Aphaenogaster miamiana&quot; &quot;Aphaenogaster fulva&quot;</span></code></pre></div>
</div>
</div>
</div>
<div id="checking-your-scientific-names" class="section level1">
<h1>Checking your scientific names</h1>
<p>You may already have a list of names, but you want to check if they all are good scientific names. <code>rphylotastic</code> calls Global Name Resolver and the Taxonomic Name Resolution Service to verify names. This will keep correct names only.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">species &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Felis silvestris&quot;</span>, <span class="st">&quot;Home sepiens&quot;</span>)
<span class="kw">taxa_resolve_names_with_otol</span>(<span class="dt">taxa =</span> species)
<span class="co">#&gt; Warning in taxa_resolve_names_with_otol(taxa = species): Fewer names were</span>
<span class="co">#&gt; found than were given; missing taxa were dropped.</span>
<span class="co">#&gt; [1] &quot;Felis silvestris&quot;</span></code></pre></div>
</div>
<div id="getting-data-from-encyclopedia-of-life" class="section level1">
<h1>Getting data from Encyclopedia of Life</h1>
<p>You can also get some images:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">some_images &lt;-<span class="st"> </span><span class="kw">species_get_image_data</span>(<span class="dt">species =</span> <span class="kw">c</span>(<span class="st">&quot;Homo sapiens&quot;</span>, <span class="st">&quot;Felis catus&quot;</span>))</code></pre></div>
<p>Or other miscelanous info:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">some_info &lt;-<span class="st"> </span><span class="kw">species_get_info</span>(<span class="dt">species =</span> <span class="kw">c</span>(<span class="st">&quot;Homo sapiens&quot;</span>, <span class="st">&quot;Felis catus&quot;</span>))</code></pre></div>
</div>
<div id="citing" class="section level1">
<h1>Citing</h1>
<p>The phylotastic manuscript is still being written.</p>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
