<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Brian O’Meara and Luna L. Sánchez-Reyes" />

<meta name="date" content="2018-12-28" />

<title>rphylotastic</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' || rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#header {
text-align: center;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; }  code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>

</head>

<body>




<h1 class="title toc-ignore">rphylotastic</h1>
<h4 class="author"><em>Brian O’Meara and Luna L. Sánchez-Reyes</em></h4>
<h4 class="date"><em>2018-12-28</em></h4>



<p><code>rphylotastic</code> connects to Phylotastic web services, which themselves wrap and process multiple other services as well as have unique services of their own.</p>
<p>You first need to install and load the package:</p>
<p>CRAN version:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" title="1"><span class="kw">install.packages</span>(<span class="st">&quot;rphylotastic&quot;</span>)</a></code></pre></div>
<p>Or development version:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" title="1"><span class="kw">library</span>(devtools)</a>
<a class="sourceLine" id="cb2-2" title="2">devtools<span class="op">::</span><span class="kw">install_github</span>(<span class="st">&quot;phylotastic/rphylotastic&quot;</span>)</a></code></pre></div>
<p>Then load the package:</p>
<div id="getting-trees" class="section level1">
<h1>Getting trees</h1>
<p>The main use for <code>rphylotastic</code> is getting phylogenetic trees for a set of taxa of interest. Trees can come from Open Tree of Life (topology only, many polytomies), Phylomatic (has branch lengths, but only plants and mammals), or NCBI taxonomy (even more polytomies than Open Tree’s tree). See also the <code>treebase</code> and <code>rotl</code> packages for getting individual trees, or <code>datelife</code> for getting chronograms (a single synthetic one or individual ones).</p>
<p>Would you like to know about north american endemic carnivorous plants and their relationship to other plants with the same behaviour? The venus flytrap (<em>Dionaea muscipula</em>), pitcher plants (<em>Sarracenia</em> genus) and the cobra lily (<em>Darlingtonia californica</em>) can only be found in North America. The popular <em>Drosera</em> sundues, <em>Pinguicula</em> butterworts and the aquatic carnivorous <em>Utricularia</em> bladderworts, have a wider range of distribution. What are their phylogenetic relationships?</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" title="1"></a>
<a class="sourceLine" id="cb3-2" title="2"><span class="co"># http://floranorthamerica.org/files/imported/Outreach/FNAfs_carnivory.pdf</span></a>
<a class="sourceLine" id="cb3-3" title="3">taxon_names &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Dionaea muscipula&quot;</span>, <span class="st">&quot;Sarracenia&quot;</span>, <span class="st">&quot;Darlingtonia californica&quot;</span>,</a>
<a class="sourceLine" id="cb3-4" title="4">          <span class="st">&quot;Drosera&quot;</span>, <span class="st">&quot;Pinguicula&quot;</span>, <span class="st">&quot;Utricularia&quot;</span>, <span class="st">&quot;Roridulaceae&quot;</span>)</a>
<a class="sourceLine" id="cb3-5" title="5">phy1 &lt;-<span class="st"> </span><span class="kw">taxa_get_otol_tree</span>(<span class="dt">taxa =</span> taxon_names)</a>
<a class="sourceLine" id="cb3-6" title="6"><span class="co"># phy1 &lt;- taxa_get_phylomatic_tree(taxa = c(&quot;elephas maximus&quot;, &quot;felis silvestris&quot;, &quot;homo sapiens&quot;))</span></a>
<a class="sourceLine" id="cb3-7" title="7"></a>
<a class="sourceLine" id="cb3-8" title="8">phy2 &lt;-<span class="st"> </span><span class="kw">taxa_get_phylomatic_tree</span>(<span class="dt">taxa =</span> taxon_names)</a>
<a class="sourceLine" id="cb3-9" title="9"></a>
<a class="sourceLine" id="cb3-10" title="10"><span class="co"># phy3 &lt;- taxa_get_taxonomic_tree(taxa = taxon_names) # this service is giving Error in open.connection(con, &quot;rb&quot;) : HTTP error 500. (2018.06.21)</span></a>
<a class="sourceLine" id="cb3-11" title="11"></a>
<a class="sourceLine" id="cb3-12" title="12">source_tree &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Open Tree of Life&quot;</span>, <span class="st">&quot;Phylomatic&quot;</span>)</a>
<a class="sourceLine" id="cb3-13" title="13"><span class="co"># source_tree &lt;- c(&quot;Open Tree of Life&quot;, &quot;Phylomatic&quot;, &quot;NCBI taxonomy&quot;)</span></a>
<a class="sourceLine" id="cb3-14" title="14"></a>
<a class="sourceLine" id="cb3-15" title="15">trees &lt;-<span class="st"> </span><span class="kw">c</span>(phy1, phy2)</a>
<a class="sourceLine" id="cb3-16" title="16"><span class="co"># trees &lt;- c(phy1, phy2, phy3)</span></a>
<a class="sourceLine" id="cb3-17" title="17"></a>
<a class="sourceLine" id="cb3-18" title="18"><span class="kw">library</span>(ape)</a>
<a class="sourceLine" id="cb3-19" title="19"><span class="kw">par</span>(<span class="dt">xpd =</span> <span class="ot">FALSE</span>, <span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>, <span class="kw">length</span>(trees)), <span class="dt">mai =</span> <span class="kw">c</span>(<span class="fl">0.1</span>, <span class="fl">0.1</span>, <span class="fl">0.5</span>, <span class="fl">0.1</span>), <span class="dt">oma =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">1</span>))</a>
<a class="sourceLine" id="cb3-20" title="20"><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(trees)){</a>
<a class="sourceLine" id="cb3-21" title="21">  <span class="kw">plot</span>(<span class="kw">ladderize</span>(trees[[i]]), <span class="dt">cex =</span> <span class="fl">0.5</span>, <span class="dt">label.offset =</span> <span class="fl">0.3</span>)</a>
<a class="sourceLine" id="cb3-22" title="22">  <span class="kw">mtext</span>(<span class="kw">paste</span>(<span class="st">&quot;from&quot;</span>, source_tree[i]), <span class="dt">cex =</span> <span class="dv">1</span>, <span class="dt">side =</span> <span class="dv">3</span>, <span class="dt">line =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb3-23" title="23">}</a>
<a class="sourceLine" id="cb3-24" title="24"><span class="kw">mtext</span>(<span class="st">&quot;Carnivorous plant tree&quot;</span>, <span class="dt">outer =</span> T, <span class="dt">side =</span> <span class="dv">3</span>, <span class="dt">cex =</span> <span class="fl">1.5</span>, <span class="dt">line =</span> <span class="dv">0</span>)</a></code></pre></div>
<p>With <code>rphylotastic</code>, you can also check if any two trees are exactly equal.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" title="1"></a>
<a class="sourceLine" id="cb4-2" title="2"><span class="kw">phylo_compare</span>(phy1, phy1)</a>
<a class="sourceLine" id="cb4-3" title="3"></a>
<a class="sourceLine" id="cb4-4" title="4"><span class="co"># since names are different in both trees, the following will be FALSE</span></a>
<a class="sourceLine" id="cb4-5" title="5"><span class="kw">phylo_compare</span>(phy1, phy2)</a></code></pre></div>
</div>
<div id="getting-scientific-names" class="section level1">
<h1>Getting scientific names</h1>
<div id="from-scratch" class="section level2">
<h2>From scratch</h2>
<p>One very cool feature of <code>rphylotastic</code> is the ability to scrape scientific names from documents or web pages. For example, you can get all latin names from a wikipedia article on primates:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" title="1"><span class="co"># names_url &lt;- url_get_scientific_names(URL = &quot;https://en.wikipedia.org/wiki/Primate&quot;) # this was failing 2018.09.17</span></a>
<a class="sourceLine" id="cb5-2" title="2"><span class="co"># length(names_url)</span></a>
<a class="sourceLine" id="cb5-3" title="3"><span class="co"># head(names_url)</span></a></code></pre></div>
<p>You can also extract all scientific names from a text you are studying, that you can insert as a character vector into R if it is short:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" title="1">names_text &lt;-<span class="st"> </span><span class="kw">text_get_scientific_names</span>(<span class="dt">text =</span> <span class="st">&quot;The domestic cat (Felis silvestris catus or Felis catus) is a small, typically furry, carnivorous mammal.&quot;</span>)</a>
<a class="sourceLine" id="cb6-2" title="2"><span class="kw">head</span>(names_text)</a>
<a class="sourceLine" id="cb6-3" title="3"><span class="co"># &gt; [1] &quot;Felis silvestris catus&quot; &quot;Felis catus&quot;</span></a></code></pre></div>
<p>or get from a file in your compputer if i=the text is too long:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" title="1">names_txt &lt;-<span class="st"> </span><span class="kw">file_get_scientific_names</span>(<span class="dt">file_name =</span> <span class="st">&quot;path/to/your/file/species.txt&quot;</span>)</a>
<a class="sourceLine" id="cb7-2" title="2">names_pdf &lt;-<span class="st"> </span><span class="kw">file_get_scientific_names</span>(<span class="dt">file_name =</span> <span class="st">&quot;path/to/your/file/species.pdf&quot;</span>)</a>
<a class="sourceLine" id="cb7-3" title="3"><span class="kw">head</span>(names_txt)</a></code></pre></div>
</div>
<div id="from-a-taxon-of-interest" class="section level2">
<h2>From a taxon of interest</h2>
<p>If you are interested in a particular group, and you want to get a tree containing all species within the group, function <code>taxon_get_species</code> will get you a vector of all species registered in Open Tree of Life taxonomy for that group. You can then use that vector to perform a tree search.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" title="1">ant_species &lt;-<span class="st"> </span><span class="kw">taxon_get_species</span>(<span class="dt">taxon =</span> <span class="st">&quot;Formicidae&quot;</span>)</a>
<a class="sourceLine" id="cb8-2" title="2"><span class="kw">length</span>(ant_species)</a>
<a class="sourceLine" id="cb8-3" title="3"><span class="co"># &gt; [1] 10883</span></a>
<a class="sourceLine" id="cb8-4" title="4"><span class="kw">head</span>(ant_species)</a>
<a class="sourceLine" id="cb8-5" title="5"><span class="co"># &gt; [1] &quot;Hypopomyrmex bombiccii&quot;  &quot;Stigmomyrmex venustus&quot;  </span></a>
<a class="sourceLine" id="cb8-6" title="6"><span class="co"># &gt; [3] &quot;Imhoffia pallida&quot;        &quot;Imhoffia nigra&quot;         </span></a>
<a class="sourceLine" id="cb8-7" title="7"><span class="co"># &gt; [5] &quot;Kotshkorkia laticeps&quot;    &quot;Poneropterus sphecoides&quot;</span></a>
<a class="sourceLine" id="cb8-8" title="8"><span class="co"># phy1 &lt;- taxa_get_otol_tree(taxa = ant_species)</span></a></code></pre></div>
<p>This function works to get all species from one taxon at a time. If you try with several higher-taxon names, it will give you the species for the first taxon only:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" title="1">more_species &lt;-<span class="st"> </span><span class="kw">taxon_get_species</span>(<span class="dt">taxon =</span> <span class="kw">c</span>(<span class="st">&quot;Formicidae&quot;</span>, <span class="st">&quot;Felidae&quot;</span>))</a>
<a class="sourceLine" id="cb9-2" title="2"><span class="co"># &gt; there is more than one name in taxon argument, only the first name will be searched</span></a>
<a class="sourceLine" id="cb9-3" title="3"><span class="kw">length</span>(more_species)</a>
<a class="sourceLine" id="cb9-4" title="4"><span class="co"># &gt; [1] 10883</span></a></code></pre></div>
<p>But you can use a handy apply function</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" title="1">more_species &lt;-<span class="st"> </span><span class="kw">rapply</span>(<span class="kw">list</span>(<span class="st">&quot;Formicidae&quot;</span>, <span class="st">&quot;Felidae&quot;</span>), taxon_get_species)</a>
<a class="sourceLine" id="cb10-2" title="2"><span class="kw">length</span>(more_species)</a>
<a class="sourceLine" id="cb10-3" title="3"><span class="co"># &gt; [1] 10982</span></a></code></pre></div>
<p>Or see <code>datelife</code> package for a wrapper of this function accepting multiple taxon names at a time.</p>
<div id="subsetting-species-names" class="section level3">
<h3>Subsetting species names</h3>
<p>Species from a taxon can also be subset by country or by genome:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" title="1">spp_from_country &lt;-<span class="st"> </span><span class="kw">taxon_get_species_from_country</span>(<span class="st">&quot;Formicidae&quot;</span>, <span class="st">&quot;Mexico&quot;</span>)</a>
<a class="sourceLine" id="cb11-2" title="2"><span class="kw">length</span>(spp_from_country)</a>
<a class="sourceLine" id="cb11-3" title="3"><span class="co"># &gt; [1] 479</span></a>
<a class="sourceLine" id="cb11-4" title="4"><span class="kw">head</span>(spp_from_country)</a>
<a class="sourceLine" id="cb11-5" title="5"><span class="co"># &gt; [1] &quot;Acanthoponera goeldii&quot; &quot;Acanthoponera minor&quot;   &quot;Tatuidris tatusia&quot;    </span></a>
<a class="sourceLine" id="cb11-6" title="6"><span class="co"># &gt; [4] &quot;Amblyopone orizabana&quot;  &quot;Prionopelta modesta&quot;   &quot;Prionopelta amabilis&quot;</span></a>
<a class="sourceLine" id="cb11-7" title="7">spp_with_genome &lt;-<span class="st"> </span><span class="kw">taxon_get_species_with_genome</span>(<span class="st">&quot;Formicidae&quot;</span>)</a>
<a class="sourceLine" id="cb11-8" title="8"><span class="kw">length</span>(spp_with_genome)</a>
<a class="sourceLine" id="cb11-9" title="9"><span class="co"># &gt; [1] 28</span></a>
<a class="sourceLine" id="cb11-10" title="10"><span class="kw">head</span>(spp_with_genome)</a>
<a class="sourceLine" id="cb11-11" title="11"><span class="co"># &gt; [1] &quot;Ooceraea biroi&quot;         &quot;Solenopsis fugax&quot;      </span></a>
<a class="sourceLine" id="cb11-12" title="12"><span class="co"># &gt; [3] &quot;Harpegnathos saltator&quot;  &quot;Dinoponera quadriceps&quot; </span></a>
<a class="sourceLine" id="cb11-13" title="13"><span class="co"># &gt; [5] &quot;Aphaenogaster picea&quot;    &quot;Aphaenogaster miamiana&quot;</span></a></code></pre></div>
</div>
</div>
</div>
<div id="checking-your-scientific-names" class="section level1">
<h1>Checking your scientific names</h1>
<p>You may already have a list of names, but you want to check if they all are good scientific names. <code>rphylotastic</code> calls Global Name Resolver (GNR) and the Taxonomic Name Resolution Service (TNRS) to verify names. This function will keep correct names only.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" title="1">species &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Delphinus delphus&quot;</span>, <span class="st">&quot;Homo sepiens&quot;</span>)</a>
<a class="sourceLine" id="cb12-2" title="2"><span class="kw">taxa_resolve_names_with_otol</span>(<span class="dt">taxa =</span> species)</a>
<a class="sourceLine" id="cb12-3" title="3"><span class="co"># &gt; Success</span></a>
<a class="sourceLine" id="cb12-4" title="4"><span class="co"># &gt; Warning in taxa_resolve_names_with_otol(taxa = species): Fewer names were</span></a>
<a class="sourceLine" id="cb12-5" title="5"><span class="co"># &gt; found than were given; missing taxa were dropped.</span></a>
<a class="sourceLine" id="cb12-6" title="6"><span class="co"># &gt; [1] &quot;Delphinus delphus&quot;</span></a></code></pre></div>
<p>Alternatively, you can use one of the many services from <code>rotl</code> and <code>taxize</code> packages to clean your names:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" title="1"><span class="kw">library</span>(taxize)</a>
<a class="sourceLine" id="cb13-2" title="2">species_gnr &lt;-<span class="st"> </span><span class="kw">gnr_resolve</span>(<span class="dt">names =</span> species)</a>
<a class="sourceLine" id="cb13-3" title="3"><span class="kw">nrow</span>(species_gnr)</a>
<a class="sourceLine" id="cb13-4" title="4"><span class="co"># &gt; [1] 32</span></a>
<a class="sourceLine" id="cb13-5" title="5"><span class="co"># it is a big table. Let&#39;s subset by unique matched names, so it is easier to choose:</span></a>
<a class="sourceLine" id="cb13-6" title="6">species_gnr[<span class="op">!</span><span class="kw">duplicated</span>(species_gnr<span class="op">$</span>matched_name),<span class="kw">c</span>(<span class="st">&quot;user_supplied_name&quot;</span>, <span class="st">&quot;matched_name&quot;</span>)]</a>
<a class="sourceLine" id="cb13-7" title="7"><span class="co"># &gt; # A tibble: 8 x 2</span></a>
<a class="sourceLine" id="cb13-8" title="8"><span class="co"># &gt;   user_supplied_name matched_name                    </span></a>
<a class="sourceLine" id="cb13-9" title="9"><span class="co"># &gt;   &lt;chr&gt;              &lt;chr&gt;                           </span></a>
<a class="sourceLine" id="cb13-10" title="10"><span class="co"># &gt; 1 Delphinus delphus  Delphinus delphus               </span></a>
<a class="sourceLine" id="cb13-11" title="11"><span class="co"># &gt; 2 Delphinus delphus  Delphinus delphus Linnaeus, 1758</span></a>
<a class="sourceLine" id="cb13-12" title="12"><span class="co"># &gt; 3 Homo sepiens       Homo sapiens Linnaeus, 1758     </span></a>
<a class="sourceLine" id="cb13-13" title="13"><span class="co"># &gt; 4 Homo sepiens       Homo sapiens                    </span></a>
<a class="sourceLine" id="cb13-14" title="14"><span class="co"># &gt; 5 Homo sepiens       Homo sapiens subsp. Denisova    </span></a>
<a class="sourceLine" id="cb13-15" title="15"><span class="co"># &gt; 6 Homo sepiens       Homo sapiens L.                 </span></a>
<a class="sourceLine" id="cb13-16" title="16"><span class="co"># &gt; 7 Homo sepiens       Homo sapiens Linnaeus 1758      </span></a>
<a class="sourceLine" id="cb13-17" title="17"><span class="co"># &gt; 8 Homo sepiens       Homo sapiens ssp. Denisova</span></a>
<a class="sourceLine" id="cb13-18" title="18"><span class="kw">tnrs</span>(<span class="dt">query =</span> species)</a>
<a class="sourceLine" id="cb13-19" title="19"><span class="co"># &gt; Calling http://taxosaurus.org/retrieve/1e78cc1bb9f28a9fc572295b13d4c8b3</span></a>
<a class="sourceLine" id="cb13-20" title="20"><span class="co"># &gt;       submittedname acceptedname    sourceid score  matchedname authority</span></a>
<a class="sourceLine" id="cb13-21" title="21"><span class="co"># &gt; 1 Delphinus delphus   Delphinium iPlant_TNRS  0.48   Delphinium        L.</span></a>
<a class="sourceLine" id="cb13-22" title="22"><span class="co"># &gt; 2      Homo sepiens              iPlant_TNRS  0.53 Hoya serpens          </span></a>
<a class="sourceLine" id="cb13-23" title="23"><span class="co"># &gt;                                     uri</span></a>
<a class="sourceLine" id="cb13-24" title="24"><span class="co"># &gt; 1 http://www.tropicos.org/Name/40023242</span></a>
<a class="sourceLine" id="cb13-25" title="25"><span class="co"># &gt; 2</span></a></code></pre></div>
<p>On the example above, species names were matched to iPlant database only. If you look at the score, it’s pretty low (&lt;0.6), so you can tell that this match is probably wrong.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" title="1"><span class="kw">library</span>(rotl)</a>
<a class="sourceLine" id="cb14-2" title="2"><span class="co"># &gt; </span></a>
<a class="sourceLine" id="cb14-3" title="3"><span class="co"># &gt; Attaching package: &#39;rotl&#39;</span></a>
<a class="sourceLine" id="cb14-4" title="4"><span class="co"># &gt; The following objects are masked from &#39;package:taxize&#39;:</span></a>
<a class="sourceLine" id="cb14-5" title="5"><span class="co"># &gt; </span></a>
<a class="sourceLine" id="cb14-6" title="6"><span class="co"># &gt;     synonyms, tax_name, tax_rank</span></a>
<a class="sourceLine" id="cb14-7" title="7">species_tnrs &lt;-<span class="st"> </span><span class="kw">tnrs_match_names</span>(<span class="dt">names =</span> species)</a>
<a class="sourceLine" id="cb14-8" title="8">species_tnrs[,<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>]</a>
<a class="sourceLine" id="cb14-9" title="9"><span class="co"># &gt;       search_string       unique_name approximate_match ott_id is_synonym</span></a>
<a class="sourceLine" id="cb14-10" title="10"><span class="co"># &gt; 1 delphinus delphus Delphinus delphis             FALSE 698438       TRUE</span></a>
<a class="sourceLine" id="cb14-11" title="11"><span class="co"># &gt; 2      homo sepiens      Homo sapiens              TRUE 770315      FALSE</span></a></code></pre></div>
<p>Package <code>brranching</code> has a very nice function that allows choosing among multiple matches. This function is actually used in phylomatic tree searches implemented via <code>rphylotastic</code> function.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" title="1"><span class="kw">library</span>(brranching)</a>
<a class="sourceLine" id="cb15-2" title="2"><span class="kw">phylomatic_names</span>(<span class="st">&quot;Byblis&quot;</span>)</a>
<a class="sourceLine" id="cb15-3" title="3"><span class="co"># [1] &quot;byblidaceae/byblis/byblis&quot;</span></a></code></pre></div>
</div>
<div id="getting-data-from-encyclopedia-of-life" class="section level1">
<h1>Getting data from Encyclopedia of Life</h1>
<p>You can also get some images:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" title="1">some_images &lt;-<span class="st"> </span><span class="kw">species_get_image_data</span>(<span class="dt">species =</span> <span class="kw">c</span>(<span class="st">&quot;Homo sapiens&quot;</span>, <span class="st">&quot;Felis catus&quot;</span>))</a></code></pre></div>
<p>Or other miscelanous info:</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" title="1">some_info &lt;-<span class="st"> </span><span class="kw">species_get_info</span>(<span class="dt">species =</span> <span class="kw">c</span>(<span class="st">&quot;Homo sapiens&quot;</span>, <span class="st">&quot;Felis catus&quot;</span>))</a></code></pre></div>
</div>
<div id="managing-your-lists-of-names" class="section level1">
<h1>Managing your lists of names</h1>
<p>You can get an phylotastic! user ID by registering with your gmail address on the portal. Then, you can get a temporary token sent to you email to help you manage your lists from the R console. Here are a few examples, but they won’t work for you since th etoken expire afer a few hours.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" title="1"><span class="kw">get_species_from_list</span>()</a>
<a class="sourceLine" id="cb18-2" title="2"><span class="kw">update_species_in_list</span>()</a>
<a class="sourceLine" id="cb18-3" title="3"><span class="kw">insert_species_in_list</span>()</a>
<a class="sourceLine" id="cb18-4" title="4"><span class="kw">replace_species_in_list</span>()</a>
<a class="sourceLine" id="cb18-5" title="5"><span class="kw">remove_species_from_list</span>()</a></code></pre></div>
</div>
<div id="citing" class="section level1">
<h1>Citing</h1>
<p>The phylotastic manuscript is still being written.</p>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
